### ISSUE TITLE
Update the macOS version used in GitHub Actions

### ISSUE THREAD

**Issue by guihkx** [nuttyartist/notes#674]
I got an email from GitHub, informing me that macOS 11, which is the version we currently use to build Notes on macOS, will be removed in about a month:

![image](https://github.com/nuttyartist/notes/assets/626206/52d4feec-d092-4116-9ad2-d46a271d399b)

We can probably perform a seamless upgrade to macOS 12, but testing the built binaries and also updating the documentation are a must, of course.

**Comment by nuttyartist** [comment]
Does that mean that future builds won't work on macOS 11?

**Comment by guihkx** [comment]
Not necessarily, I believe.

It's been a while since I built anything on macOS, but I do remember the `CMAKE_OSX_DEPLOYMENT_TARGET` CMake option, which [we actually use](https://github.com/nuttyartist/notes/blob/94351b6b87906224f2ba7d60e318af5c55a1df45/CMakeLists.txt#L12-L14) to target our builds for macOS 10.15 (although Qt 6 itself doesn't support that macOS version, which makes it useless, lol).

So we can probably update `CMAKE_OSX_DEPLOYMENT_TARGET` to `11.0`, and then test the build on macOS 11.

**Comment by nuttyartist** [comment]
Haha, sounds good! I wonder how we could get our hands on a macOS 11 machine? Mine is 12. Maybe there are services for that. I've seen somewhere (mentioned in a Qt bug report) about one but forgot to save it.

**Comment by guihkx** [comment]
> I wonder how we could get our hands on a macOS 11 machine

Every macOS-related work I did for Notes, was done using macOS 11 running via QEMU on Linux, so I know that's one possibility. Even though animations were very slow, since it lacked any support for GPU acceleration, it was at least "fast enough" to build, test and debug some issues...

Unfortunately, I don't have that VM set up anymore (I'm running low on disk space, lol), and I don't exactly remember how I set it up, but I probably used [this amazing tool](https://github.com/quickemu-project/quickemu).

Though I'm afraid it's Linux-only, unfortunately.
